test/e2e/targets/azureeventhubs/main.go

```diff
@@ -193,9 +292,9 @@ var _ = Describe("Azure Event Hubs target", func() {
 
 					// NOTE: The payload will be a stringified version of the CloudEvent
 					Expect(payload).To(ContainSubstring(string(event.Data())))
-					Expect(payload).To(ContainSubstring("type: " + event.Type()))
-					Expect(payload).To(ContainSubstring("source: " + event.Source()))
-					Expect(payload).To(ContainSubstring("id: " + event.ID()))
+					Expect(payload).ToNot(ContainSubstring("type: " + event.Type()))
```

Comment:
Echoing my previous comment: considering the purpose of this PR we should be able to assume there is no need to compare to whole payload to substrings anymore.
We could compare the Event Hub message to `helloEvent.Data()` though.

Reply:
In this case I'd like to ensure that none of the CloudEvent data made it through without deserializing the string. Granted could just still to a full `Equals` instead of `ContainsSubstring`.

Reply:
If you compare Message.Data to HelloEvent.Data you can be sure there are no leftovers of context attributes.